examples/GTEx/extract_tile_PW_GTEx.py

```diff
@@ -0,0 +1,275 @@
+import argparse
+import os
+import time
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+import requests
+from sklearn.model_selection import train_test_split
+from tqdm import tqdm
+
+from histolab.slide import SlideSet
+from histolab.tiler import RandomTiler
+
+URL_ROOT = "https://brd.nci.nih.gov/brd/imagedownload"
+
+
+def download_wsi_gtex(dataset_dir: str, sample_ids: List[str]) -> None:
+    """Download into ``dataset_dir`` all the GTEx WSIs corresponding to ``sample_ids``
+
+    Parameters
+    ----------
+    dataset_dir : str
+        Path where to save the WSIs
+    sample_ids : List[str]
+        List of GTEx WSI ids
+    """
+    for sample_id in tqdm(sample_ids):
+        if f"{sample_id}.svs" not in os.listdir(dataset_dir):
+
+            request = requests.get(f"{URL_ROOT}/{sample_id}")
+            with open(
+                os.path.join(dataset_dir, f"{sample_id}.svs"), "wb"
+            ) as output_file:
+                output_file.write(request.content)
+
+            time.sleep(np.random.randint(60, 100))
```

Comment:
this line seems fishy to me? what's the scope of that `time.sleep`?
Furthermore download files with requests where each file can be huge I would suggest to use the `stream` param along with the download by chunks ... something like:

```python
def download_file(url):
    filename = url.split('/')[-1]
    with requests.get(url, stream=True) as r:
        r.raise_for_status()
        with open(filename, 'wb') as f:
            for chunk in r.iter_content(chunk_size=8192): 
                # If you have chunk encoded response uncomment if
                # and set chunk_size parameter to None.
                #if chunk: 
                f.write(chunk)
    return filename
```

Reply:
The sleep seems necessary because GTEx servers don't seem very performant... when downloading without sleep 70/255 slides were HTML with 'Service not available' 🤷‍♀️

Reply:
It seems no API service available i suppose ... If you need a sleep there means that no Api infrastructure is provided and this could be a really mess. IMHO this script won't work for each ID because there's no validator in the endpoint ... so it seems very weak. Do we have time to take another look and try to find alternative service for this kind of image we're searching for?